ext/oci8/oci8.c

```diff
@@ -2373,6 +2395,11 @@ static int php_oci_connection_close(php_oci_connection *connection TSRMLS_DC)
 		pefree(connection->client_id, connection->is_persistent);
 	}
 #endif /* HAVE_OCI8_DTRACE */
+	
+	if (connection->taf_callback) {
+		pefree(connection->taf_callback, connection->is_persistent);
```

Comment:
connection->taf_callback needs to be set to NULL, otherwise a memory leak may occur. While disconnecting from a normal connection, php_oci_connection_close() gets called before php_oci_disable_taf_callback(). If connection->taf_callback is not set to NULL, it gets reallocated and filled with PHP_OCI_TAF_DISABLE_CALLBACK in the last call of php_oci_disable_taf_callback(). However, php_oci_connection_close() has already been called at this point and connection->taf_callback will not be freed. Thus, a memory leak occurs.